frontend/src/libs/components/button/button.tsx

```diff
@@ -26,25 +28,40 @@ const Button: React.FC<Properties> = ({
   className,
   icon,
   isLabelVisuallyHidden = false,
-}: Properties) => (
-  <button
-    onClick={onClick}
-    type={type}
-    disabled={isDisabled}
-    className={getValidClassNames(
+  to,
+}: Properties) => {
+  const content = (
+    <>
+      {icon && <Icon iconName={icon} />}
+      <span
+        className={getValidClassNames(
+          isLabelVisuallyHidden && 'visually-hidden',
+        )}
+      >
+        {label}
+      </span>
+    </>
+  );
+
+  const buttonProperties = {
+    onClick,
+    type,
+    disabled: isDisabled,
+    className: getValidClassNames(
       styles['button'],
       styles[style],
       styles[size],
       className,
-    )}
-  >
-    {icon && <Icon iconName={icon} />}
-    <span
-      className={getValidClassNames(isLabelVisuallyHidden && 'visually-hidden')}
-    >
-      {label}
-    </span>
-  </button>
-);
+    ),
+  };
+
+  return to ? (
+    <Link to={to} className={buttonProperties.className}>
+      {content}
+    </Link>
+  ) : (
+    <button {...buttonProperties}>{content}</button>
```

Comment:
```suggestion
    <button <pass_props_explicitly> >{content}</button>
```

Reply:
U mean like
```
<button
      onClick={onClick}
      type={type}
      disabled={isDisabled}
      className={getValidClassNames(
        styles['button'],
        styles[style],
        styles[size],
        className,
      )}
    >
      {content}
    </button>
```

Reply:
Yep

Reply:
but for example `className` we can pass to the link too. so create a constant for it and use for both

Reply:
And in Link component I should do the same way, so that I gonna have like 2 calls of getValidClassNames with the same output, so can I create for that an const or it doesn't matter?

Reply:
k